streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -294,7 +304,10 @@ public static StreamThread create(final InternalTopologyBuilder builder,
                                       final long cacheSizeBytes,
                                       final StateDirectory stateDirectory,
                                       final StateRestoreListener userStateRestoreListener,
-                                      final int threadIdx) {
+                                      final int threadIdx,
+                                      final ShutdownErrorHook shutdownErrorHook,
+                                      final StreamsUncaughtExceptionHandler streamsUncaughtExceptionHandler,
+                                      final AtomicInteger assignmentErrorCode) {
```

Comment:
It doesn't look like this needs to be shared outside of this thread. It seems like it just needs to be shared between the StreamThread and its Consumer?

Reply:
You are right it seems that it is not necessary